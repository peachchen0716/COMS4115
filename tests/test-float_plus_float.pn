def foo(float x, float y): float {
	return x + y;
}

foo(3.0, 4.0);