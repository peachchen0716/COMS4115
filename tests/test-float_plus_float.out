; ModuleID = 'Pyni'
source_filename = "Pyni"

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.1 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.2 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.3 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.4 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.5 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1

declare i32 @printf(i8*, ...)

define double @foo(double %x, double %y) {
entry:
  %x1 = alloca double
  store double %x, double* %x1
  %y2 = alloca double
  store double %y, double* %y2
  %x3 = load double, double* %x1
  %y4 = load double, double* %y2
  %float_binop = fadd double %x3, %y4
  ret double %float_binop
}

define i32 @main() {
entry:
  %foo_result = call double @foo(double 3.000000e+00, double 4.000000e+00)
  ret i32 0
}
